tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/MatchingStrategies.kt

```diff
@@ -44,17 +45,22 @@ open class UsageCompatibility: AttributeCompatibilityRule<Usage> {
 
         when {
             requested == null -> compatible()
-            requested == Usage.JAVA_API && (provided == Usage.JAVA_API || provided == KotlinNativeUsage.KLIB) -> {
+            requested == provided -> compatible()
+            requested in supportedRequestedUsages && provided in supportedProvidedUsages -> {
                 compatible()
             }
-            requested == provided -> compatible()
         }
     }
+
+    companion object {
+        val supportedRequestedUsages = listOf(KotlinUsages.KOTLIN_API, Usage.JAVA_API)
+        val supportedProvidedUsages = listOf(KotlinUsages.KOTLIN_API, Usage.JAVA_API, KotlinNativeUsage.KLIB)
+    }
 }
 
 open class UsageDisambiguation: AttributeDisambiguationRule<Usage> {
     override fun execute(details: MultipleCandidatesDetails<Usage>): Unit = with(details) {
-        val usagePriority = listOf(Usage.JAVA_API, KotlinNativeUsage.KLIB)
+        val usagePriority = listOf(KotlinUsages.KOTLIN_API, Usage.JAVA_API, KotlinNativeUsage.KLIB)
```

Comment:
By the way, there's a built-in function for order-like disambiguation: [link](https://docs.gradle.org/current/javadoc/org/gradle/api/attributes/AttributeMatchingStrategy.html#ordered-java.util.Comparator-)

Reply:
(UPD: fixed the link in the comment above)

Reply:
Thanks! :)